src/foam/blob/HttpServletRequestBlob.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.blob;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.BufferedInputStream;
+import java.nio.ByteBuffer;
+
+public class HttpServletRequestBlob
+    extends foam.blob.AbstractBlob
+{
+  protected long size_;
+  protected long pos_ = 0;
+  protected HttpServletRequest request_;
+
+
+  public HttpServletRequestBlob(HttpServletRequest request) {
+    this.request_ = request;
+    this.size_ = request.getContentLengthLong();
+  }
+
+  @Override
+  public Buffer read(Buffer buffer, long offset) {
+    try {
+      if ( offset != pos_ ) {
+        throw new Exception("Offset does not match stream position");
```

Comment:
Should be RuntimeException?

Reply:
Shouldn't matter cause it should be caught by the catch block regardless.